prime-router/src/main/kotlin/fhirengine/translation/TranslationSchemaManager.kt

```diff
@@ -0,0 +1,91 @@
+            val input = sourceContainer.getBlobClient(inputDirectoryPath + "input.fhir")
+            val inputBundle = FhirTranscoder.decode(input.toString())
+
+            val transformBlob = blobs.filter {
+                it.currentBlobItem.name.startsWith(inputDirectoryPath) &&
+                it.currentBlobItem.name.endsWith(".yml")
+            }
+            val transformBlobName = transformBlob[0].currentBlobItem.name
+
+            when (schemaType) {
+                SchemaType.FHIR -> {
+                    val configSchema = ConfigSchemaReader.fromFile(
+                        blobEndpoint + "/" + blobContainerInfo.containerName + "/" + transformBlobName,
+                        null,
+                        FhirTransformSchema::class.java,
+                        blobContainerInfo
+                    )
+                    val transformedInputBundle = if (configSchema is FhirTransformSchema) {
+                        FhirTransformer(configSchema).transform(inputBundle)
+                    } else {
+                        throw InternalServerErrorException("Transform schema $transformBlob not a FhirTransform")
+                    }
+                    val output = sourceContainer.getBlobClient(inputDirectoryPath + "output.fhir").downloadContent()
+                    // Unfortunately need to do this to get it to be the same format.
+                    val expectedOutput = FhirTranscoder.decode(output.toString())
+                    // .equals is required here == does not compare it properly, despite what IntelliJ says
+                    if (transformedInputBundle.equals(expectedOutput)) {
+                        BlobAccess.logger.error("Validation failed for transform $transformBlobName")
+                        return false
+                    }
+                }
+                SchemaType.HL7 -> {
+                    val converterSchema = ConfigSchemaReader.fromFile(
+                        blobEndpoint + "/" + blobContainerInfo.containerName + "/" + transformBlobName,
+                        null,
+                        ConverterSchema::class.java,
+                        blobContainerInfo
+                    )
+                    val hl7Transform = FhirToHl7Converter(
+                        converterSchema as ConverterSchema,
+                        false,
+                        null,
+                        ConstantSubstitutor(),
+                        null
+                    ).convert(inputBundle)
+                    val output = sourceContainer.getBlobClient(inputDirectoryPath + "output.hl7").downloadContent()
+                    if (!hl7Transform.toString().trim().equals(output.toString().trim())) {
+                        BlobAccess.logger.error("Validation failed for transform $transformBlobName")
+                        return false
+                    }
+                }
+            }
+        }
+
+        return true
```

Comment:
I thought we discussed returning a data object that contains results of the validations? That way when running from the CLI, we can print out the validation results instead of making people dig through logs.